utils/dockerfile_generator.py

```diff
@@ -0,0 +1,185 @@
+
+            if update:
+                with dockerfile_path.open() as dockerfile:
+                    dockerfile = dockerfile.read()
+
+                    # Dockerfiles with this string will be frozen; skip.
+                    if '### MANUAL ###' in dockerfile:
+                        continue
+                
+            else:
+                dockerfile_path.touch()
+                
+            
+            service_config_paths = [Path(x) for x in sorted(analyzer.iterdir()) if x.suffix == '.json']
+            
+            # Grab the first JSON file found-- they _should_ have the same relevant properties anyway
+            json_config = service_config_paths[0]
+            
+            if json_config.exists():
+
+                dockerfile_contents = ["### AUTOGEN ###\n# THIS FILE IS UPDATED BY `utils/dockerfile_builder.py`\n# DO NOT EDIT IT DIRECTLY\n"]
+
+                config = loads(json_config.open().read())
+                process_path = Path('..', 'analyzers', config['command']).resolve()
+
+                # Determine baseImage for Dockerfile
+                if 'baseImage' in config:
+                    baseImage = config['baseImage']
+                else:
+                    # Analyze source code as determined by config file
+                    with process_path.open() as process_src:
+                        source_code = process_src.read()
+                        dockerfile_contents.append("# Guessing base image from source code shebang")
+                        if '#!/usr/bin/env python\n' in source_code:
+                            # If we're building every image, we should save disk space w/ alpine
+                            baseImage = 'python:2-alpine'
+                            is_python = True
+                        elif '#!/usr/bin/env python3\n' in source_code:
+                            # If we're building every image, we should save disk space w/ alpine
+                            baseImage = 'python:3-alpine'
+                            is_python = True
+                        
+                        # TODO: Add more runtime shebangs! ex. go, rust, ruby
+
+                        # Default out to ubuntu
+                        else:
+                            baseImage = 'ubuntu'
+
+
+                if 'alpine' in baseImage:
+                    is_alpine = True
+
+                dockerfile_contents.append('FROM {}\n'.format(baseImage))
+
+
+                # Include alpine-specific dependencies
+                if is_alpine:
+
+                    # Using a set to prevent duplicate install operations
+                    alpine_dependencies = set()
+                    if is_python:
+                        requirements_path = analyzer / 'requirements.txt'
+                        if requirements_path.exists():
+                            requirements = requirements_path.open().read()
+                            
+                            if 'yara-python' in requirements:
```

Comment:
This is the real meat here... 'dependency analysis' might be a bit generous considering how iterative the approach here is. Thankfully a lot of these use the same, common reversing libraries.